common/firmware/i2c/i2c_comms.cpp

```diff
@@ -13,12 +15,12 @@ using namespace i2c;
  */
 I2C::I2C(HAL_I2C_HANDLE handle) : handle{handle} {}
 
-auto I2C::master_transmit(uint16_t dev_address, uint8_t *data, uint16_t size,
+auto I2C::master_transmit(uint8_t* data, uint16_t size, uint16_t dev_address,
                           uint32_t timeout) -> bool {
     return hal_i2c_master_transmit(handle, dev_address, data, size, timeout);
```

Comment:
This is definitely outside the scope of this PR, and possibly unnecessary, but it looks like the task is using the blocking `HAL_I2C_Master_Transmit` function. Moving to the `_IT` versions of the HAL functions and implementing a synchronization system to pause the task during I2C transactions would free up some CPU time for other tasks to run during a transaction. Not sure if this is an optimization you need though.

Reply:
yep was planning on transitioning to using either the `IT` or DMA versions of I2C transactions in a follow-up

Reply:
Doesn't the blocking version actually spin on HAL_Delay which is, i'm pretty sure, rtos-aware